src/main/java/uk/gov/companieshouse/web/accounts/interceptor/CompanyAccountsDataStateInterceptor.java

```diff
@@ -0,0 +1,180 @@
+
+                    companyAccountsDataStates = new CompanyAccountsDataStates();
+                }
+
+            }
+
+            // Get the state from the request session
+            CompanyAccountsDataState companyAccountsDataState = (CompanyAccountsDataState) request.getSession().getAttribute(STATE_REQUEST_ATTRIBUTE);
+
+            if (isApprovalSubmission(request)) {
+                // Remove the state for this company accounts id - it's not needed any more
+                companyAccountsDataStates.getCompanyAccountsDataStatesMap().remove(companyAccountsId);
+            } else {
+                // Remove the oldest 'state' if more than 5 are present to prevent bloating the JWT
+                if (companyAccountsDataStates.getCompanyAccountsDataStatesMap().size() >= 5) {
+                    removeOldestState(companyAccountsDataStates);
+                }
+                // Update / insert the state object on the state map, using the company accounts id as the key
+                companyAccountsDataStates
+                        .getCompanyAccountsDataStatesMap().put(companyAccountsId, companyAccountsDataState);
+            }
+
+            try {
+                // Convert the states object back to a JWT and set it on the cookie
+                String token = tokenManager.createJWT(companyAccountsDataStates);
+                stateCookie.setValue(token);
+                stateCookie.setPath("/");
+
+                // Add the cookie to the response to save it
+                response.addCookie(stateCookie);
+
+            } catch (JsonProcessingException e) {
+                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+            }
+        }
+    }
+
+    private Cookie getStateCookie(HttpServletRequest request) {
+
+        return Arrays.stream(request.getCookies())
+                .filter(cookie -> cookie.getName().equalsIgnoreCase(STATE_COOKIE_NAME))
+                .findFirst()
+                .orElse(null);
+    }
+
+    private CompanyAccountsDataState createNewState() {
+
+        CompanyAccountsDataState companyAccountsDataState = new CompanyAccountsDataState();
+
+        AccountingPolicies accountingPolicies = new AccountingPolicies();
+        companyAccountsDataState.setAccountingPolicies(accountingPolicies);
+
+        return companyAccountsDataState;
+    }
+
+    private String getCompanyAccountsIdFromRequest(HttpServletRequest request) {
+
+        Matcher matcher = COMPANY_ACCOUNTS_REGEX.matcher(request.getRequestURI());
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+
+    private boolean isApprovalSubmission(HttpServletRequest request) {
+
+        Matcher matcher = APPROVAL_REGEX.matcher(request.getRequestURI());
+        return matcher.find() && request.getMethod().equalsIgnoreCase("POST");
+    }
+
+    private void removeOldestState(CompanyAccountsDataStates companyAccountsDataStates) {
```

Comment:
I think this method could use javadoc as even though I believe the method name is descriptive enough, it would be beneficial to understand why you would use this method - similar to the comment on line 115.

Reply:
Done 👍